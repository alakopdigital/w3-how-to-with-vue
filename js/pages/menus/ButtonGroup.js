webpackJsonp([30],{"./node_modules/babel-loader/lib/index.js!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./src/components/menus/ButtonGroup.vue":function(e,o,s){"use strict";o.a={name:"button-group",functional:!0,props:{isFullwidth:Boolean,isVertical:Boolean},render:function(e,o){return e("div",{class:["button-group",{"is-fullwidth":o.props.isFullwidth,"is-vertical":o.props.isVertical}]},o.children)}}},"./node_modules/babel-loader/lib/index.js!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./src/pages/menus/ButtonGroup.vue":function(e,o,s){"use strict";var t=s("./src/components/menus/ButtonGroup.vue");o.a={name:"button-group-page",data:function(){return{intro:'a "button group" with CSS',buttons:["Apple","Samsung","Sony"],examples:[["``` html",'<div class="btn-group">',"  <button>Apple</button>","  <button>Samsung</button>","  <button>Sony</button>","</div>","```"],["``` css",".btn-group button {","  background-color: #4CAF50; /* Green background */","  border: 1px solid green; /* Green border */","  color: white; /* White text */","  padding: 10px 24px; /* Some padding */","  cursor: pointer; /* Pointer/hand icon */","  float: left; /* Float the buttons side by side */","}","",".btn-group button:not(:last-child) {","  border-right: none; /* Prevent double borders */","}","","/* Clear floats (clearfix hack) */",".btn-group:after {",'  content: "";',"  clear: both;","  display: table;","}","","/* Add a background color on hover */",".btn-group button:hover {","  background-color: #3e8e41;","}","```"],["``` html","\x3c!-- Three buttons in a group --\x3e",'<div class="btn-group" style="width:100%">','  <button style="width:33.3%">Apple</button>','  <button style="width:33.3%">Samsung</button>','  <button style="width:33.3%">Sony</button>',"</div>","","\x3c!-- Four buttons in a group --\x3e",'<div class="btn-group" style="width:100%">','  <button style="width:25%">Apple</button>','  <button style="width:25%">Samsung</button>','  <button style="width:25%">Sony</button>','  <button style="width:25%">HTC</button>',"</div>","```"]]}},components:{ButtonGroup:t.a}}},'./node_modules/css-loader/index.js!./node_modules/vue-loader/lib/style-compiler/index.js?{"vue":true,"id":"data-v-4422938c","scoped":false,"hasInlineConfig":false}!./node_modules/sass-loader/lib/loader.js!./node_modules/vue-loader/lib/selector.js?type=styles&index=0!./src/pages/menus/ButtonGroup.vue':function(e,o,s){o=e.exports=s("./node_modules/css-loader/lib/css-base.js")(!1),o.push([e.i,"",""])},'./node_modules/css-loader/index.js!./node_modules/vue-loader/lib/style-compiler/index.js?{"vue":true,"id":"data-v-f07c75ea","scoped":false,"hasInlineConfig":false}!./node_modules/sass-loader/lib/loader.js!./node_modules/vue-loader/lib/selector.js?type=styles&index=0!./src/components/menus/ButtonGroup.vue':function(e,o,s){o=e.exports=s("./node_modules/css-loader/lib/css-base.js")(!1),o.push([e.i,".button-group{display:flex}.button-group.is-fullwidth{width:100%}.button-group.is-vertical{flex-direction:column}.button-group button{background-color:#4caf50;border:1px solid green;color:#fff;padding:10px 24px;cursor:pointer;flex:1}.button-group button:hover{background-color:#3e8e41}.button-group button:not(:last-child){border-right:none}",""])},'./node_modules/vue-loader/lib/template-compiler/index.js?{"id":"data-v-4422938c","hasScoped":false,"buble":{"transforms":{}}}!./node_modules/vue-loader/lib/selector.js?type=template&index=0!./src/pages/menus/ButtonGroup.vue':function(e,o,s){"use strict";var t=function(){var e=this,o=e.$createElement,s=e._self._c||o;return s("page",{staticClass:"button-group-page",attrs:{title:"Button Group",intro:e.intro}},[s("template",{slot:"header"},[s("h2",{staticClass:"title"},[e._v("Horizontal Button Groups")]),e._v(" "),s("p",[e._v("Group a series of buttons together on a single line in a button group:")])]),e._v(" "),s("sample",[s("button-group",e._l(e.buttons,function(o){return s("button",{key:o,domProps:{textContent:e._s(o)}})}))],1),e._v(" "),s("br"),e._v(" "),s("sample",[s("button-group",{attrs:{"is-fullwidth":""}},e._l(e.buttons,function(o){return s("button",{key:o,domProps:{textContent:e._s(o)}})}))],1),e._v(" "),s("template",{slot:"examples"},[s("hr"),e._v(" "),s("h2",{staticClass:"title"},[e._v("How To Create a Button Group")]),e._v(" "),s("step",{attrs:{no:"1",html:""}}),e._v(" "),s("example-code",{attrs:{code:e.examples[0]}}),e._v(" "),s("step",{attrs:{no:"2",css:""}}),e._v(" "),s("example-code",{attrs:{code:e.examples[1]}}),e._v(" "),s("hr"),e._v(" "),s("h2",{staticClass:"title"},[e._v("Justified / Full-width Button Group:")]),e._v(" "),s("example-code",{attrs:{code:e.examples[2]}})],1)],2)},l=[],n={render:t,staticRenderFns:l};o.a=n},'./node_modules/vue-style-loader/index.js!./node_modules/css-loader/index.js!./node_modules/vue-loader/lib/style-compiler/index.js?{"vue":true,"id":"data-v-4422938c","scoped":false,"hasInlineConfig":false}!./node_modules/sass-loader/lib/loader.js!./node_modules/vue-loader/lib/selector.js?type=styles&index=0!./src/pages/menus/ButtonGroup.vue':function(e,o,s){var t=s('./node_modules/css-loader/index.js!./node_modules/vue-loader/lib/style-compiler/index.js?{"vue":true,"id":"data-v-4422938c","scoped":false,"hasInlineConfig":false}!./node_modules/sass-loader/lib/loader.js!./node_modules/vue-loader/lib/selector.js?type=styles&index=0!./src/pages/menus/ButtonGroup.vue');"string"==typeof t&&(t=[[e.i,t,""]]),t.locals&&(e.exports=t.locals);s("./node_modules/vue-style-loader/lib/addStylesClient.js")("87b82776",t,!0,{})},'./node_modules/vue-style-loader/index.js!./node_modules/css-loader/index.js!./node_modules/vue-loader/lib/style-compiler/index.js?{"vue":true,"id":"data-v-f07c75ea","scoped":false,"hasInlineConfig":false}!./node_modules/sass-loader/lib/loader.js!./node_modules/vue-loader/lib/selector.js?type=styles&index=0!./src/components/menus/ButtonGroup.vue':function(e,o,s){var t=s('./node_modules/css-loader/index.js!./node_modules/vue-loader/lib/style-compiler/index.js?{"vue":true,"id":"data-v-f07c75ea","scoped":false,"hasInlineConfig":false}!./node_modules/sass-loader/lib/loader.js!./node_modules/vue-loader/lib/selector.js?type=styles&index=0!./src/components/menus/ButtonGroup.vue');"string"==typeof t&&(t=[[e.i,t,""]]),t.locals&&(e.exports=t.locals);s("./node_modules/vue-style-loader/lib/addStylesClient.js")("1c1a41e2",t,!0,{})},"./src/components/menus/ButtonGroup.vue":function(e,o,s){"use strict";function t(e){s('./node_modules/vue-style-loader/index.js!./node_modules/css-loader/index.js!./node_modules/vue-loader/lib/style-compiler/index.js?{"vue":true,"id":"data-v-f07c75ea","scoped":false,"hasInlineConfig":false}!./node_modules/sass-loader/lib/loader.js!./node_modules/vue-loader/lib/selector.js?type=styles&index=0!./src/components/menus/ButtonGroup.vue')}var l=s("./node_modules/babel-loader/lib/index.js!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./src/components/menus/ButtonGroup.vue"),n=s("./node_modules/vue-loader/lib/component-normalizer.js"),d=t,u=n(l.a,null,!1,d,null,null);o.a=u.exports},"./src/pages/menus/ButtonGroup.vue":function(e,o,s){"use strict";function t(e){s('./node_modules/vue-style-loader/index.js!./node_modules/css-loader/index.js!./node_modules/vue-loader/lib/style-compiler/index.js?{"vue":true,"id":"data-v-4422938c","scoped":false,"hasInlineConfig":false}!./node_modules/sass-loader/lib/loader.js!./node_modules/vue-loader/lib/selector.js?type=styles&index=0!./src/pages/menus/ButtonGroup.vue')}Object.defineProperty(o,"__esModule",{value:!0});var l=s("./node_modules/babel-loader/lib/index.js!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./src/pages/menus/ButtonGroup.vue"),n=s('./node_modules/vue-loader/lib/template-compiler/index.js?{"id":"data-v-4422938c","hasScoped":false,"buble":{"transforms":{}}}!./node_modules/vue-loader/lib/selector.js?type=template&index=0!./src/pages/menus/ButtonGroup.vue'),d=s("./node_modules/vue-loader/lib/component-normalizer.js"),u=t,r=d(l.a,n.a,!1,u,null,null);o.default=r.exports}});
//# sourceMappingURL=ButtonGroup.js.map?id=17c61375a854f1e3719c