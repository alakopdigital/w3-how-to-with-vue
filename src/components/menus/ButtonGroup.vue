<script>
export default {
  name: 'button-group',
  functional: true,
  props: {
    isFullwidth: Boolean
  },
  render(h, ctx) {
    return h(
      'div',
      { class: ['button-group', { 'is-fullwidth': ctx.props.isFullwidth }] },
      ctx.children
    )
  }
}
</script>

<style lang="scss">
.button-group {
  display: flex;

  &.is-fullwidth {
    width: 100%;
  }

  button {
    background-color: #4CAF50; /* Green background */
    border: 1px solid green; /* Green border */
    color: white; /* White text */
    padding: 10px 24px; /* Some padding */
    cursor: pointer; /* Pointer/hand icon */
    flex: 1;

    &:hover {
      /* Add a background color on hover */
      background-color: #3e8e41;
    }

    &:not(:last-child) {
      /* Prevent double borders */
      border-right: none;
    }
  }
}
</style>
