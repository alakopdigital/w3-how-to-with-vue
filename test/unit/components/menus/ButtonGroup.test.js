import { mount, shallow, renderer } from '#/unit/utils'
import ButtonGroup from '@/components/menus/ButtonGroup.vue'

describe('ButtonGroup.vue', () => {
  it('render default', () => {
    const wrap = mount(ButtonGroup)

    expect(wrap.hasClass('button-group')).toBe(true)
    expect(wrap.isEmpty()).toBe(true)

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('props#is-fullwidth', () => {
    const wrap = mount(ButtonGroup, {
      context: {
        props: { isFullwidth: true }
      }
    })

    expect(wrap.hasClass('is-fullwidth')).toBe(true)

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('props#is-vertical', () => {
    const wrap = mount(ButtonGroup, {
      context: {
        props: { isVertical: true }
      }
    })

    expect(wrap.hasClass('is-vertical')).toBe(true)

    renderer.renderToString(wrap.vm, (err, str) => {
      if (err) console.error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
