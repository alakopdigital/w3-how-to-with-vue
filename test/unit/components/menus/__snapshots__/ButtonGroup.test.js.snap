// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup.vue props#is-fullwidth 1`] = `<div class="button-group is-fullwidth"></div>`;

exports[`ButtonGroup.vue render default 1`] = `<div class="button-group"></div>`;
